src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveKerrSchild.hpp

```diff
@@ -0,0 +1,225 @@
+            Requires<std::is_same<ScalarFieldSolution, Dummy>::value> = nullptr>
+  explicit ScalarWaveKerrSchild(
+      double mass, std::array<double, volume_dim> dimensionless_spin,
+      std::array<double, volume_dim> center,
+      std::unique_ptr<MathFunction<1>> profile,
+      const OptionContext& context = {}) noexcept;
+
+  // Enable or disable this constructor based on ScalarFieldSolution
+  template <typename Dummy = ScalarWave::Solutions::PlaneWave<volume_dim>,
+            Requires<std::is_same<ScalarFieldSolution, Dummy>::value> = nullptr>
+  explicit ScalarWaveKerrSchild(
+      double mass, std::array<double, volume_dim> dimensionless_spin,
+      std::array<double, volume_dim> center,
+      std::array<double, volume_dim> wave_vector,
+      std::array<double, volume_dim> wave_center,
+      std::unique_ptr<MathFunction<1>> profile,
+      const OptionContext& context = {}) noexcept;
+
+  explicit ScalarWaveKerrSchild(CkMigrateMessage* /*unused*/) noexcept {}
+
+  ScalarWaveKerrSchild() = default;
+  ScalarWaveKerrSchild(const ScalarWaveKerrSchild& /*rhs*/) = delete;
+  ScalarWaveKerrSchild& operator=(const ScalarWaveKerrSchild& /*rhs*/) = delete;
+  ScalarWaveKerrSchild(ScalarWaveKerrSchild&& /*rhs*/) noexcept = default;
+  ScalarWaveKerrSchild& operator=(ScalarWaveKerrSchild&& /*rhs*/) noexcept =
+      default;
+  ~ScalarWaveKerrSchild() = default;
+
+  // Tags specific to the solution
+  using scalar_wave_tags = tmpl::list<Pi, Phi<volume_dim>, Psi>;
+  using scalar_wave_dt_tags = db::wrap_tags_in<::Tags::dt, scalar_wave_tags>;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using spacetime_tags = tmpl::list<
+      gr::Tags::Lapse<DataVector>, ::Tags::dt<gr::Tags::Lapse<DataVector>>,
+      DerivLapse, gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>,
+      DerivShift,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>,
+      DerivSpatialMetric, gr::Tags::SqrtDetSpatialMetric<DataVector>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using tags = tmpl::append<spacetime_tags, scalar_wave_tags>;
+
+  /// Returns curved scalar wave variables
+  tuples::tagged_tuple_from_typelist<scalar_wave_tags> variables(
```

Comment:
can you update the interface to conform with the way the interfaces work for all our other analytic solutions? Will recently updated Kerr-Schild to conform to the standard way and I'd rather not have new ones that need to be updated in the future since that leads to people copying the incorrect interface.

Reply:
Updated the interface, now any arbitrary number / combination of tags involved can be retrieved. This is cleaner too.